src/libs/auth.js

```diff
@@ -327,32 +327,50 @@ authStore.subscribe(withErrorReporting('Error checking registration', async (sta
 authStore.subscribe(withErrorReporting('Error checking TOS', async (state, oldState) => {
   if (!oldState.isSignedIn && state.isSignedIn) {
     const tosDetails = await Ajax().User.getTermsOfServiceDetails()
-    if (_.isNull(tosDetails)) {
-      const termsOfService = {
-        isGracePeriodEnabled: undefined,
-        currentVersion: undefined,
-        userAcceptedVersion: undefined,
-        userAcceptedTos: false,
-        userNeedsToAcceptTos: true,
-      }
-      authStore.update(state => ({ ...state, termsOfService }))
-    } else {
-      const userAcceptedTos = !_.isUndefined(tosDetails.currentVersion) ?
-        tosDetails.currentVersion === tosDetails.userAcceptedVersion : undefined
-      const userNeedsToAcceptTos = !_.isUndefined(userAcceptedTos) ?
-        (!userAcceptedTos && !tosDetails.isGracePeriodEnabled) || _.isUndefined(tosDetails.userAcceptedVersion) : false
-      const termsOfService = {
-        isGracePeriodEnabled: tosDetails.isGracePeriodEnabled,
-        currentVersion: tosDetails.currentVersion,
-        userAcceptedVersion: tosDetails.userAcceptedVersion,
-        userAcceptedTos,
-        userNeedsToAcceptTos,
-      }
-      authStore.update(state => ({ ...state, termsOfService }))
-    }
+    const termsOfService = parseToSDetails(tosDetails)
+    authStore.update(state => ({ ...state, termsOfService }))
   }
 }))
 
+const parseToSDetails = tosDetails => {
+  if (_.isNull(tosDetails)) {
+    return {
+      isGracePeriodEnabled: undefined,
+      currentVersion: undefined,
+      userAcceptedVersion: undefined,
+      userAcceptedTos: false,
+      userNeedsToAcceptTos: true,
+    }
+  } else {
+    // IF user has accepted latest version of ToS
+    // THEN let them use Terra, do not show them the ToS popup
+    // ELSE IF user accepted ANY prior version of ToS and Grace Period is enabled
+    // THEN let them use Terra, do show them the ToS popup
+    // ELSE                ///////// IF user has no accepted ToS version OR Grace Period is disabled
```

Comment:
Formatting?

Reply:
@gpolumbo-broad formatted it this way on purpose

Reply:
Ok. It seemed like maybe the IF was intended to be on the next line.

Reply:
I still think this would be more legible without the spaces and `//////` or with `IF user ...` on the next line.